.github/workflows/test.yml

```diff
@@ -168,13 +168,23 @@ jobs:
     continue-on-error: true
     strategy:
       matrix:
-        version: ["1.71.1", beta, "nightly-2024-05-23"]
+        version: ["current_msrv", beta, "nightly"]
     steps:
       - uses: actions/checkout@v4
 
+      - name: Get Rust version to install
+        id: rust_version
+        run: |
+          if [ "${{ matrix.version }}" = "current_msrv" ]; then
+            msrv=$(cat Cargo.toml | grep rust-version | sed 's/.* = "//; s/"//')
```

Comment:
This could fail in various ways, for example if whitespace is removed around the equal sign or if someone adds a comment (or other string) anywhere in the Cargo.toml file that matches on 'rust-version'.

I'd recommend using instead something like:

```
msrv=$(sed -En '/^rust-version.*=/ { s/.*\"(.*?)\"/\1/; p }' < Cargo.toml)
```

Reply:
Yes, that's definitely a valid point

In general, it should happen quire rare in contrast with MSRV bump flow - because for a new maintainer it's still the same flow: bump msrv in cargo -> forget to update CI -> **fail of CI in PR** -> **figure out** where you should to bump it

Subjectively speaking, I still don't find this a complex option. Because once you changed something in `Cargo.toml` in a way it breaks this script - you will get failed CI run. With pretty clear error message about not existing rustc version (in most of cases).  

But I didn't try to force anyone to use it if there is no desire.